### DISCUSSION TITLE
Encoding error on Search Terms report

### DISCUSSION THREAD

**Discussion by vgarciasouza** [denisneuf/python-amazon-ad-api#147] [Q&A]
Hello, and first of all, thank for your help.

On the "spSearchTerm" report I get an encoding error when downloading if there was a "searchterm" with special caracters in the day (eg. "searchTerm": "狗狗厕所")

Here's my code line where raises the error:

`        ads_client.download_report(
            url=url, format='json', file=str(report_file.absolute()), encoding='utf-8')`

and this is what I get;

`  \utils\ads_reports.py", line 109, in download_report_to_csv
    ads_client.download_report(
  File "\ad_api\api\reports.py", line 70, in download_report
    return self._download(self, params=kwargs, headers={'User-Agent': self.user_agent})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  \ad_api\base\client.py", line 156, in _download
    fo.write(read_data.decode('UTF-8'))
  \Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'charmap' codec can't encode characters in position 15032-15035: character maps to <undefined>`

**Comment by KritikoAlex** [comment]
Hello @vgarciasouza,
I am currently experiencing the same issue. Could you resolve the issue? I am trying to download the report file with format set to 'raw' but I can't seem to find the correct decoding strategy in order to work with chinese/cyrillic letters.

Kind regards,
Alex

**Comment by vgarciasouza** [comment]
Hello @KritikoAlex ,

Today, I successfully resolved the error that occurred when downloading as 'json'.

In the file client.py, on line 153, it's necessary to add encoding='UTF-8', so the line should be:

`fo = open(name + ".json", 'w', encoding='UTF-8')`
